qt/widgets/common/inc/MantidQtWidgets/Common/UserInputValidator.h

```diff
@@ -77,14 +87,14 @@ class DLLExport UserInputValidator {
   bool checkWorkspaceGroupIsValid(QString const &groupName, QString const &inputType, bool silent = false);
 
   /// Add a custom error message to the list.
-  void addErrorMessage(const QString &message, bool silent = false);
+  void addErrorMessage(const std::string &message, bool const silent = false) override;
 
   /// Sets a validation label
   void setErrorLabel(QLabel *errorLabel, bool valid);
 
   /// Returns an error message which contains all the error messages raised by
   /// the check functions.
-  QString generateErrorMessage();
+  std::string generateErrorMessage() const override;
   /// Checks to see if all input is valid
   bool isAllInputValid();
 
```

Comment:
Is there any reason to not include 'serErrorLabel' and 'isAllInputValid' in the interface? I guess that would be useful in the future to mock the input user validator fully

Reply:
Ok, no problem, that sounds good. I just thought they were currently used in the code but if not it is better to add them later as you said